Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -738,8 +742,12 @@ bool RenderLayerCompositor::updateCompositingLayers(CompositingUpdateType update
         needHierarchyUpdate |= layersChanged;
     }
 
-#if !LOG_DISABLED
-    if (compositingLogEnabled() && isFullUpdate && (needHierarchyUpdate || needGeometryUpdate)) {
+#if !LOG_DISABLED || ENABLE(RESOURCE_USAGE)
+    if (
+#if !ENABLE(RESOURCE_USAGE)
+            compositingLogEnabled() &&
+#endif
+            isFullUpdate && (needHierarchyUpdate || needGeometryUpdate)) {
```

Comment:
Don't do this, please.
Can't you do something like this in compositingLogEnabled() to avoid breaking the ifs that way?

#if !LOG_DISABLED || ENABLE(RESOURCE_USAGE)
static inline bool compositingLogEnabled()
{
#if ENABLE(RESOURCE_USAGE)
    return true;
#endif
    return LogCompositing.state == WTFLogChannelOn;
}
#endif

The rest looks ok.